tests/rpc/RPCServer.test.ts

```diff
@@ -741,8 +741,341 @@ describe(`${RPCServer.name}`, () => {
+  test('default timeout', async () => {
+    const ctxProm = promise<ContextTimed>();
+    class TestHandler extends RawHandler {
+      public handle(_input, _connectionInfo, _ctx): ReadableStream<Uint8Array> {
+        ctxProm.resolveP(_ctx);
+        // Do nothing, expecting timeout
+        let controller: ReadableStreamController<Uint8Array>;
+        const stream = new ReadableStream<Uint8Array>({
+          start: (_controller) => {
+            controller = _controller;
+          },
+        });
+        _ctx.signal.addEventListener('abort', () => {
+          controller!.error(Error('ending'));
+        });
+        return stream;
+      }
+    }
+    const rpcServer = await RPCServer.createRPCServer({
+      manifest: {
+        testMethod: new TestHandler({}),
+      },
+      defaultTimeout: 100,
+      logger,
+    });
+    const [outputResult, outputStream] = rpcTestUtils.streamToArray();
+    const stream = rpcTestUtils.messagesToReadableStream([
+      {
+        jsonrpc: '2.0',
+        method: 'testMethod',
+        params: null,
+      },
+      {
+        jsonrpc: '2.0',
+        method: 'testMethod',
+        params: null,
+      },
+    ]);
+    const readWriteStream: ReadableWritablePair = {
+      readable: stream,
+      writable: outputStream,
+    };
+    rpcServer.handleStream(readWriteStream, {} as ConnectionInfo);
+    const ctx = await ctxProm.p;
+    expect(ctx.timer.delay).toEqual(100);
+    await ctx.timer;
+    expect(ctx.signal.reason).toBeInstanceOf(rpcErrors.ErrorRpcTimedOut);
+    await expect(outputResult).toReject();
+    await rpcServer.destroy();
+  });
+  test('default timeout before handler selected', async () => {
+    const rpcServer = await RPCServer.createRPCServer({
+      manifest: {},
+      defaultTimeout: 100,
+      logger,
+    });
+    const readWriteStream: ReadableWritablePair = {
+      readable: new ReadableStream({
+        // Ignore
+        cancel: () => {},
+      }),
+      writable: new WritableStream({
+        // Ignore
+        abort: () => {},
+      }),
+    };
+    rpcServer.handleStream(readWriteStream, {} as ConnectionInfo);
+    // With no handler we can only check alive connections through the server
+    // @ts-ignore: kidnap protected property
+    const activeStreams = rpcServer.activeStreams;
+    for await (const [prom] of activeStreams.entries()) {
+      await prom;
+    }
+    await rpcServer.destroy();
+  });
+  test('handler overrides timeout', async () => {
+    {
+      const waitProm = promise();
+      const ctxShortProm = promise<ContextTimed>();
+      class TestMethodShortTimeout extends UnaryHandler {
+        timeout = 25;
+        public async handle(input: JSONValue, _, _ctx): Promise<JSONValue> {
+          ctxShortProm.resolveP(_ctx);
+          await waitProm.p;
+          return input;
+        }
+      }
+      const ctxLongProm = promise<ContextTimed>();
+      class TestMethodLongTimeout extends UnaryHandler {
+        timeout = 100;
```

Comment:
Overriding is setting the property?

Reply:
Yes, the alternative is to set it during construction.

Reply:
The `timeout` property for `abstract class Handler` should have a docblock explaining what it is used for when used with downstream classes.

Reply:
A comment has been added.